erc20-price/src/lib.rs

```diff
@@ -188,3 +190,158 @@ fn store_chainlink_price(
+                    .unwrap();
+
+            let pair_created_event = PairCreatedEvent {
+                token0: Some(Erc20Token {
+                    address: token0_asset.address,
+                    name: token0_asset.name,
+                    symbol: token0_asset.symbol,
+                    decimals: token0_asset.decimals,
+                }),
+                token1: Some(Erc20Token {
+                    address: token1_asset.address,
+                    name: token1_asset.name,
+                    symbol: token1_asset.symbol,
+                    decimals: token1_asset.decimals,
+                }),
+                pair: Hex(event.pair.clone()).to_string(),
+                factory: Hex(log.address()).to_string(),
+            };
+
+            output.set(
+                0,
+                keyer::pair_info_key(&pair_created_event.pair),
+                &pair_created_event,
+            );
+        }
+    }
+}
+
+#[substreams::handlers::store]
+fn store_uniswap_price(
+    block: eth::Block,
+    chainlink_prices: StoreGetProto<Erc20Price>,
+    store: StoreGetProto<PairCreatedEvent>,
+    output: StoreSetProto<Erc20Price>,
+) {
+    for log in block.logs() {
+        if let Some(event) = pair::events::Sync::match_and_decode(log) {
+            let pair_address = Hex(log.address()).to_string();
+
+            if let Some(pair) = store.get_last(keyer::pair_info_key(&pair_address)) {
+                let reserve0 = event
+                    .reserve0
+                    .to_decimal(pair.token0.clone().unwrap().decimals);
+                let reserve1 = event
+                    .reserve1
+                    .to_decimal(pair.token1.clone().unwrap().decimals);
+
+                // TODO: Add a check for mininmum liquidity threshhold.
+
+                match pair.token0.clone().unwrap().address.as_str() {
+                    address if utils::STABLE_COINS.contains(&address) => {
+                        let token_price = reserve0.clone() / reserve1.clone();
+
+                        let erc20price = Erc20Price {
+                            token: pair.token1.clone(),
+                            price_usd: token_price.to_string(),
+                            block_number: block.number,
+                            source: 2,
+                        };
+
+                        output.set(
+                            0,
+                            keyer::uniswap_asset_key(&pair.token0.clone().unwrap().address),
+                            &erc20price,
+                        );
+                    }
+                    address if utils::TOKENS.get("ETH").unwrap().eq(&address) => {
+                        let eth_price = match chainlink_prices.get_last(keyer::chainlink_asset_key(
+                            &pair.token0.clone().unwrap().address,
+                        )) {
+                            Some(price) => BigDecimal::from_str(price.price_usd.as_str()).unwrap(),
```

Comment:
Think we should change the field ```string price_usd = 2;``` to use a BigDecimal field instead to avoid string parsing

Reply:
yeah sounds good, I will change this field to BigDecimal type.

Reply:
@robinbernon what type would `BigDecimal` be?